.github/workflows/release.yml

```diff
@@ -181,12 +181,18 @@ jobs:
         ref: ${{ inputs.commit_sha }}
         path: smithy-rs
         token: ${{ secrets.RELEASE_AUTOMATION_BOT_PAT }}
+        fetch-depth: 0
     - name: Generate release artifacts
       uses: ./smithy-rs/.github/actions/docker-build
       with:
         action: generate-smithy-rs-release
     - name: Download all artifacts
       uses: ./smithy-rs/.github/actions/download-all-artifacts
+    # This step is not idempotent, as it pushes release artifacts to the `smithy-rs-release-1.x.y` branch. However,
+    # if this step succeeds but a subsequent step fails, retrying the release workflow is "safe" in that it does not
+    # create any inconsistent states; this step would simply fail because the release branch would be ahead of `main`
+    # due to previously pushed artifacts.
+    # To successfully retry a release, revert the commits in the release branch that pushed the artifacts.
```

Comment:
Does this mean revert all of the commits to the release branch so it is on the same commit as main, or only the commits from the failed execution of the release? We should probably document this in our internal ops docs. 

And could we get some of the context from this comment `echo`ed from the script if it does fail, I think it would be beneficial for it to be output with the failure so whoever runs the release doesn't have to go searching through the action definition to find this comment.

Reply:
> Does this mean revert all of the commits to the release branch so it is on the same commit as main, or only the commits from the failed execution of the release?

Yes

> We should probably document this in our internal ops docs.

We documented it in our ops docs two weeks ago.

> And could we get some of the context from this comment echoed from the script if it does fail

Good idea, given the amount of changes, we'll do that in a subsequent PR (this error does happen today and this PR is not making it worse on that front).

Reply:
> Does this mean revert all of the commits to the release branch

Yes

> We should probably document this in our internal ops docs

We documented that in our internal docs two weeks ago (thanks @drganjoo)

> And could we get some of the context from this comment echoed from the script if it does fail

That's a reasonable idea (we need to ensure that the error is displayed no matter how the script errors out)

Reply:
> Does this mean revert all of the commits to the release branch

Yes

> We should probably document this in our internal ops docs

We documented that in our internal docs two weeks ago (thanks @drganjoo)

> And could we get some of the context from this comment echoed from the script if it does fail

Hmm, the error message must be placed in all subsequent scripts after this step (putting the error message in this step won't be as helpful because if this step fails, no harm has NOT been done). This error happens today, and this PR is on par with where we are (at least we have this explanatory comment). We'll keep eye on the need to echo some messages, but I'd defer it to a future PR.

Reply:
> Does this mean revert all of the commits to the release branch

Yes

> We should probably document this in our internal ops docs

We documented that in our internal docs two weeks ago (thanks @drganjoo)

> And could we get some of the context from this comment echoed from the script if it does fail

Hmm, the error message must be placed in all subsequent scripts after this step (putting the error message in this step won't be as helpful because if this step fails, no harm has been done). This error happens today, and this PR is on par with where we are (at least we have this explanatory comment). We'll keep eye on the need to echo some messages, but I'd defer it to a future PR.